src/html-media-source.js

```diff
@@ -159,6 +159,19 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       });
     };
 
+    this.onPlayerSeeking_ = (event) => {
+      this.sourceBuffers.forEach((sourceBuffer) => {
+        if (sourceBuffer.transmuxer_) {
+          let range = this.player_.buffered();
+          let current = this.player_.currentTime();
+
+          if (current >= range.end(0) || current <= range.start(0)) {
```

Comment:
I think this is all that would need to be done from contrib-hls https://github.com/mjneil/videojs-contrib-hls/commit/6cfdb6b32950a88908954230d47d6d554484b5a2